test/e2e/framework/viperconfig/viperconfig.go

```diff
@@ -0,0 +1,117 @@
+
+func init() {
+	flag.StringVar(&viperConfig, "viper-config", "", "The name of the viper config i.e. 'e2e' will read values from 'e2e.json' locally.  All e2e parameters can also be configured in such a file. May contain a path and may or may not contain the file suffix.")
+}
+
+// ViperizeFlags checks whether a configuration file was specified, reads it, and updates
+// the configuration variables accordingly. Must be called after framework.HandleFlags()
+// and before framework.AfterReadingAllFlags().
+func ViperizeFlags() {
+	if viperConfig != "" {
+		viper.SetConfigName(filepath.Base(viperConfig))
+		viper.AddConfigPath(filepath.Dir(viperConfig))
+		if err := viper.ReadInConfig(); err != nil {
+			// If the user specified a file suffix, the Viper won't
+			// find the file because it always appends its known set
+			// of file suffices. Therefore try once more without
+			// suffix.
+			ext := filepath.Ext(viperConfig)
+			if ext != "" && err.Error() == viperFileNotFound {
+				viper.SetConfigName(filepath.Base(viperConfig[0 : len(viperConfig)-len(ext)]))
+				err = viper.ReadInConfig()
+			}
+			if err != nil {
+				// If parsing a config was requested, then it
+				// must succeed. This catches syntax errors
+				// and "file not found". Unfortunately error
+				// messages are sometimes hard to understand,
+				// so try to help the user a bit.
+				switch err.Error() {
+				case viperFileNotFound:
+					panic(fmt.Sprintf("Viper config file %s not found or not using a supported file format", viperConfig))
+				default:
+					panic(fmt.Sprintf("Error parsing Viper config file %s: %s", viperConfig, err))
+				}
+			}
+		}
+
+		// Expose all command line flags also as Viper keys.
+		pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+		pflag.Parse()
+		viper.BindPFlags(pflag.CommandLine)
+
+		// Update all flag values with values found via
+		// Viper. We do this ourselves instead of calling
+		// viper.Unmarshal(&TestContext) because that would
+		// only update values in the main context and miss
+		// values stored in the context of individual tests.
+		viperUnmarshal()
+	}
+}
+
+// viperUnmarshall updates all command line flags with the corresponding values found
+// via Viper, regardless whether the flag value is stored in TestContext, some other
+// context or a local variable.
+func viperUnmarshal() {
+	flag.VisitAll(func(f *flag.Flag) {
+		if viper.IsSet(f.Name) {
+			// In contrast to viper.Unmarshal(), values
+			// that have the wrong type (for example, a
+			// list instead of a plain string) will not
+			// trigger an error here. This could be fixed
+			// by checking the type ourselves, but
+			// probably isn't worth the effort.
+			//
+			// "%v" correctly turns bool, int, strings into
+			// the representation expected by flag, so those
+			// can be used in config files. Plain strings
+			// always work there, just as on the command line.
+			value := viper.Get(f.Name)
+			str := fmt.Sprintf("%v", value)
```

Comment:
possibly combine multiple variables in a single line:
```
fmt.Sprintf("%v", viper.Get(f.Name))
```

Reply:
Done.